### PR TITLE
[add] unit test

### PR DESCRIPTION
**Pull request by kazukazuinaina**
#### Requirements (please check off with 'x')

- [x] I have read the [Contributing Guidelines](https://github.com/ryanoasis/vim-devicons/blob/master/contributing.md)
- [x] I have read or at least glanced at the [FAQ](https://github.com/ryanoasis/vim-devicons#faq--troubleshooting)
- [x] I have read or at least glanced at the [Wiki](https://github.com/ryanoasis/vim-devicons/wiki)

#### What does this Pull Request (PR) do?

I have introduced unit tests on the devicons global api. 
Static analysis by vint and execution of test code improve the quality of quality assurance.

#### How should this be manually tested?

1. Install and setting vim-themis

> https://github.com/thinca/vim-themis

2. run command below in devicons root dir

```bash
themis test
```

#### Any background context you can provide?

> https://github.com/kazukazuinaina/vim-devicons/runs/653510174

This shows the result of running the test on the forked repository.

##### using tool

- github-actions
- vim-themis

### CODE REVIEW

**Comment by kazukazuinaina** [comment]
What do you think?
@ryanoasis

**Comment by ryanoasis** [comment]
This is great! I think you are my hero, I've wanted to add at least _some_ unit tests for years now :blush: . I think it looks great so far, I want to take a deeper and longer look at it though

File: test/filetype.vim [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+function! s:suite.FileNodesDefaultSymbol()
+  call s:assert.equals('', WebDevIconsGetFileTypeSymbol())
+endfunction
+
+function! s:suite.VimSymbol()
+  call s:assert.equals('', WebDevIconsGetFileTypeSymbol('.vimrc'))
+endfunction
+
+function! s:suite.LicenseSymbol()
+  call s:assert.equals('', WebDevIconsGetFileTypeSymbol('LICENSE'))
```
Comment (ryanoasis) [inline test/filetype.vim:17]: Would all the `WebDevIconsGetFileTypeSymbol` asserts be separate functions or would they get grouped together? Maybe this is just a first step as you alluded to...

**Comment by kazukazuinaina** [comment]
@ryanoasis 
Can I merge this once?
Or should I add more test cases in this PR?

**Comment by ryanoasis** [comment]
Can I merge this once?
> I think it looks good and by now I trust you. I will try to look at it in the next few days.

Or should I add more test cases in this PR?
> I feel like it's a good start and we can move forward with at least these sanity tests that you have so far :smile:

File: test/filetype.vim (hunk 1 above)
Comment (kazukazuinaina) [reply test/filetype.vim:17]: These can be grouped or separated individually.

**Comment by kazukazuinaina** [comment]
@ryanoasis 
Thank you😄

**Comment by kazukazuinaina** [comment]
@ryanoasis 

I made a few corrections.
This allows unit testing on some versions of Vim.

**Comment by ryanoasis** [comment]
No need to squash unless you feel the changes are all too similar. These commits seem fine to me though

**Comment by kazukazuinaina** [comment]
OK, I got it.

**Comment by kazukazuinaina** [comment]
ping @ryanoasis

**Comment by ryanoasis** [comment]
I think it looks good! I don't understand all the changes but looks like you know what you're doing and it's a great step forward for adding more testing that I've always wanted to add.

I'm going to be pretty unresponsive for some time. Feel free to merge this and anything else you deem fit. Just @ me though (like you've already been doing) if you have doubts but I won't guarantee a timely response :sweat_smile:

**Comment by kazukazuinaina** [comment]
Thanks, @ryanoasis .
I have merged this test

**Pull request merged**